src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -49,10 +50,32 @@ public byte[] getMessage() {
         return message;
     }
 
+    public void addReplacementOperation(int startIndex, int endIndex, int maskLength, byte maskByte) {
+        ReplacementOperation replacementOperation = new ReplacementOperation(startIndex, endIndex, maskLength, maskByte);
+        replacementOperations.add(replacementOperation);
+        replacementOperationsTotalDifference += replacementOperation.difference();
+    }
+
+    public List<ReplacementOperation> getReplacementOperations() {
+        return replacementOperations;
+    }
+
+    public int getReplacementOperationsTotalDifference() {
+        return replacementOperationsTotalDifference;
+    }
+
     // for debugging purposes, shows the current state of message traversal
-    public String peek() {
+    @Override
+    public String toString() {
```

Comment:
Does it really make sense to call this toString? peek() or debugPrint() seemed more reasonable to me

Reply:
The main reason is that debugger calls to toString by default, so you don't have to setup a watch, but no strong opinion here

Reply:
Me neither, so lets keep it like this